### PR TITLE
Feature/us15/entry view edit mode

### PR DESCRIPTION
**Pull request by dimitriosxmi**
## Please take a moment to review this ❤️ 😊

**Goal**: [US15](https://github.com/dimitriosxmi/ArtistsReferenceOrganizer/issues/22) **TL; DR**: Existing entries can now be editted through the `Entry form` which can be accessed through `Edit entry button` which can be found in the entry view page.

[Deployment](https://artists-reference-organizer-gzcjouov2-dimitriosxmi.vercel.app/)

```diff
+ Add PenIcon svg
! Update components/svgs/index.js
# Exports the new svg.

+ Add EditEntryButton component
# Basically almost the same as the NewEntryButton component.

! Update BackButton component
# Now behaves diffently when called with editMode.

! Update entry.js api page
# Handles the PATCH method for updating an entry by id.

! Update NewEntryForm component
# Renamed to EntryForm.
# Adapted to now behave diffently when it
# receives the editEntryData through props:
# Fills all input fields with the received data.
# Submit button styling and Icon changes.
# Submitting makes a PATCH instead of POST request.

! Update new-entry.js page
# Updated import name of the previously NewEntryForm component.

! Update EntryDetails component
# Now uses EditEntryButton component.
# Now uses EntryForm component.

! Update [entryId].js page
# Uses the editMode and setEditMode state variable.
# Passes new props to BackButton and EntryDetails components
```

### CODE REVIEW

File: components/EditEntryButton/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import styled from "styled-components";
+import Link from "next/link";
+// SVGs
+import { EntryFileIcon, PenIcon } from "../svgs";
+
+const EditEntryButton = ({ toggleEditMode }) => {
+  return (
+    <StyledButton onClick={() => toggleEditMode()}>
```
Comment (ryliecc) [inline components/EditEntryButton/index.js:8]: just fyi, if you dont use special attributes like here you can usually just put in the function without making an arrow function out of it, like this: `onClick={toggleEditMode}`

File: components/EntryDetails/index.js [hunk 2]
```diff
@@ -28,25 +31,35 @@ const EntryDetails = () => {
...
-        {entryTags.map((tag) => (
-          <StyledTagCard key={tag._id}>{tag.tagName}</StyledTagCard>
-        ))}
-      </StyledTagContainer>
-      {/* Description */}
-      <StyledText borderTop bold>
-        Description:
-      </StyledText>
-      <StyledText>{entryDescription}</StyledText>
+      {/* When editmode is toggled,
```
Comment (ryliecc) [inline components/EntryDetails/index.js:34]: Don't leave comments in your pull requests, that is very bad practice (I have been told from my head coach lol, been there before myself)

**Review by ryliecc — APPROVED** [review]
Looks good to me 😌 The comments are just minor suggestions to keep in mind for the future. 🤓

File: components/EntryDetails/index.js (hunk 2 above)
Comment (dimitriosxmi) [reply components/EntryDetails/index.js:34]: Heyooo @ryliecc 
Thank you very much for your time and for your review! 😊 
I am absolutely certain (as an experience developer and a Documentations fan) comments that explain the code in human language are absolutely welcome in any piece of code. 

What is bad practice, is when code is commented out and left there. i.e. unused commented code.

**Pull request merged**